### DISCUSSION TITLE
Safari - multiple redirects when blocking 3rd party cookies (setCookie "domain" parameter)

### DISCUSSION THREAD

**Discussion by MGLL** [keycloak/keycloak#26219] [Q&A]
Hello,

We have an issue where **Safari** (_yes_) consider the cookies generated by Keycloak under a sub-domain as 3rd party cookies. As we are looking to block 3rd party cookies on our platform, it cause multiple redirects from our FE to KC.
According to our FE lead dev, we don't have this issue on Chrome & Firefox.

We currently have this kind of setup:
- _auth.example.com_ -> Keycloak
- _app.example.com_ -> Platform (Secured)
- _example.com_ -> Showcase site

On the FE we use React with https://github.com/authts/react-oidc-context and we have check-sso on init (onLoad). 

While checking the source code of Keycloak, I saw that there is a `domain` parameter on the `setCookie` methods of Keycloak, but it's always set to `null`. 

**Is there a reason for that ?**

We are currently wondering if allowing a wild card domain (like _.example.com_) or base domain (example.com) in `setCookie` could solve this issue.
But as we are using Keycloak in a container and the `null` value is hardcoded, it seems we have no way to verify this theory.

**Comment by jonkoops** [comment]
This sounds like a bug, we should be setting the domain. Can you log this as an issue?

**Reply by pedroigor** [reply]
This is probably because of how those browsers handle an empty domain. I'm not 100% sure what the specs say about it but if Chrome and Firefox are working is because they are setting the domain as the request host if not set in the cookie ...

**Reply by pedroigor** [reply]
But yeah, we should probably consider setting a domain and perhaps also make this configurable to whether you want to bind it to subdomains. It should be more reliable than leave it empty and rely on different behaviors from different browsers.

**Comment by MGLL** [comment]
Sure, I will create an issue, thanks for the response 👍

**Reply by MGLL** [reply]
Here is the issue: https://github.com/keycloak/keycloak/issues/26240
